build/template-Build-run-tests-sign.yml

```diff
@@ -0,0 +1,213 @@
+  displayName: 'dotnet --list-sdks'
+  inputs:
+    command: custom
+    custom: '--list-sdks'
+
+- task: CmdLine@2
+  displayName: 'Run VerifyResourceUsage.pl'
+  inputs:
+    filename: perl
+    arguments: '$(Agent.BuildDirectory)\s\$(WilsonSourceDirectory)src\VerifyResourceUsage.pl'
+
+- powershell: |
+    regedit /s .\build\strongNameBypass.reg
+  displayName: 'Strong Name Bypass'
+
+- task: DotNetCoreCLI@2
+  displayName: Build
+  inputs:
+    command: 'build'
+    projects: '$(WilsonSourceDirectory)Product.proj'
+    arguments: '/r:True /p:Configuration=$(BuildConfiguration) /p:Platform="Any CPU" /verbosity:m /p:SourceLinkCreate=true'
+
+- task: PowerShell@2
+  displayName: 'Run Tests'
+  inputs:
+    targetType: filePath
+    filePath: ./$(WilsonSourceDirectory)runTests.ps1
+    arguments: '-buildType $(BuildConfiguration)'
+
+- task: CopyFiles@2
+  displayName: 'Copy Files to: [staging]\ProductBinaries'
+  inputs:
+    SourceFolder: $(WilsonSourceDirectory)src
+    Contents: |
+      **\bin\$(BuildConfiguration)\**\Microsoft.IdentityModel.*.dll
+      **\bin\$(BuildConfiguration)\**\Microsoft.IdentityModel.*.pdb
+      **\bin\$(BuildConfiguration)\**\System.IdentityModel.Tokens.Jwt.dll
+      **\bin\$(BuildConfiguration)\**\System.IdentityModel.Tokens.Jwt.pdb
+    TargetFolder: '$(Build.ArtifactStagingDirectory)\ProductBinaries'
+
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-policheck.PoliCheck@2
+  displayName: 'Run PoliCheck'
+  inputs:
+    targetType: F
+    result: PoliCheck.xml
+    optionsFC: 0
+    optionsXS: 0
+    optionsHMENABLE: 0
+  condition: and(succeeded(), eq(variables['PipelineType'], 'legacy'))
+
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@3
+  displayName: 'Run CredScan'
+  inputs:
+    outputFormat: pre
+    suppressionsFile: 'build/credscan-exclusion.json'
+    debugMode: false
+  condition: and(succeeded(), eq(variables['PipelineType'], 'legacy'))
+
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-roslynanalyzers.RoslynAnalyzers@2
+  displayName: 'Run Roslyn Analyzers'
+  condition: and(eq(variables['TargetNet8'], 'False'), eq(variables['PipelineType'], 'Legacy'))
+
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-binskim.BinSkim@4
+  displayName: 'Run BinSkim'
+  inputs:
+    AnalyzeSymPath: '$(Build.ArtifactStagingDirectory)\ProductBinaries'
+    AnalyzeVerbose: true
+    AnalyzeHashes: true
+  condition: and(succeeded(), eq(variables['PipelineType'], 'legacy'))
+
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@3
+  displayName: 'Publish Security Analysis Logs'
+  continueOnError: true
+
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
+  displayName: 'Post SDL Analysis'
+  continueOnError: true
+
+#Sign Wilson 7x task group
+- template: template-sign-wilson.yaml
```

Comment:
Why did you split them in templates? I do not see these 2 templates being used more than once...

Reply:
I didn't split them. this is how it was from the beginning. I can do some additional refactoring to make things simpler.